scalafix/rules/src/main/scala/fix/Scalaz2cats_v1_0_0.scala

```diff
@@ -123,23 +137,36 @@ case class MigrateValidationNel(index: SemanticdbIndex) extends SemanticRule(ind
         ctx.replaceTree(t, importer"cats.syntax.validated._".syntax)
       case t @ importer"scalaz.syntax.apply._" =>
         ctx.replaceTree(t, importer"cats.syntax.apply._".syntax)
-      case t @ importer"scalaz.{..$ips}" =>
+      case importer"scalaz.{..$ips}" =>
         ips.collect {
           case NonEmptyListScalaz(i: Importee) =>
             ctx.removeImportee(i) + ctx.addGlobalImport(importer"cats.data.NonEmptyList")
         }.asPatch
       case Term.ApplyInfix(_, cartesianBuilders(op: Term.Name), _, _) =>
         replaceOpWithComma(ctx, op)
+      case Term.ApplyInfix(lhs, cartesianOps(_), _, _) =>
+        wrapInParensIfNeeded(ctx, lhs)
       case Term.Apply(t @ Term.ApplyInfix(_, cartesianBuilders(_), _, _), _) =>
         ctx.addRight(t, ".mapN")
       case Term.Apply(NonEmptyListScalaz(t), _) =>
-        val x = ctx.tokenList.next(t.tokens.last)
         ctx.addRight(t, ".of")
-    }.asPatch + (if (ctx.tree.collect {
-      case t @ Term.Select(_, successNel(_) | failureNel(_))
+    }.asPatch + ctx.replaceSymbols(cartesianAppliesRenames.toSeq: _*) + (if (ctx.tree.collect {
+      case Term.Select(_, successNel(_) | failureNel(_))
         if !ctx.tree.contains(scalazValidationSyntaxImport) => ()
-    }.length > 0) ctx.addGlobalImport(catsValidatedSyntaxImport) else Patch.empty)
+    }.nonEmpty) ctx.addGlobalImport(catsValidatedSyntaxImport) else Patch.empty)
   }
+
+  //maybe we should refactor this to some helper object ???
```

Comment:
I think I'll eventually move this to scalafix itself: feels like a useful method to have in the core library. Let's keep it here for now.